src/fabric/src/fabric2_fdb.erl

```diff
@@ -158,10 +161,35 @@ transactional(#{tx := {erlfdb_transaction, _}} = Db, Fun) ->
     end).
 
 
+enable_grv_cache() ->
+    put(?PDICT_GRV, pending).
+
+
+reset_grv_cache() ->
+    case get(?PDICT_GRV) of
+        undefined ->
+            ok;
+        _ ->
+            put(?PDICT_GRV, pending)
+    end.
+
+
 do_transaction(Fun, LayerPrefix) when is_function(Fun, 1) ->
     Db = get_db_handle(),
     try
         erlfdb:transactional(Db, fun(Tx) ->
+            case get(?PDICT_GRV) of
+                undefined ->
+                    ok;
+                pending ->
+                    put(?PDICT_GRV, {future, erlfdb:get_read_version(Tx)});
```

Comment:
Could we use `erlfdb:wait(erlfdb:get_read_version(Tx))` directly here as opposed to going through a future?

The GRV read happens as the first thing in every transaction and it should be cached. Otherwise reading the future in a second transaction might fail if the first transaction is reset to due an error or times out. From what I recall with the exception for watches, futures have to be waited on in the same transaction which created them.

Reply:
I think you misread this. get_read_version/1 returns a future, I simply store the future. The expectation is that the future is ready inbetween this call and the next. doing the wait early would be a negative (or neutral) performance hit?

Reply:
In practice it would work, and the future would be ready by the time the next transaction runs, however, the code is still structured in a way where we are waiting on a future from a previous transaction in the new transaction, and except for watches, that doesn't seem correct. 

One approach might be to call `erlfdb:wait(GrvFuture)` after the transaction function executes to ensure we get the future after the main transaction body ran, but we are still in the same transaction, and by that time the result will be there.  But I think simply waiting on it to start with might be ok from a performance point of view, since each transaction has to get the grv anyway.